sandbox/modules/identity/src/main/java/org/opensearch/identity/SecurityRestFilter.java

```diff
@@ -0,0 +1,157 @@
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.util.concurrent.ThreadContext;
+import org.opensearch.rest.BytesRestResponse;
+import org.opensearch.rest.RestChannel;
+import org.opensearch.rest.RestHandler;
+import org.opensearch.rest.RestRequest;
+import org.opensearch.rest.RestStatus;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class SecurityRestFilter {
+
+    protected final Logger log = LogManager.getLogger(this.getClass());
+    private final ThreadContext threadContext;
+    private final Settings settings;
+    private final Path configPath;
+
+    public SecurityRestFilter(final ThreadPool threadPool, final Settings settings, final Path configPath) {
+        super();
+        this.threadContext = threadPool.getThreadContext();
+        this.settings = settings;
+        this.configPath = configPath;
+    }
+
+    /**
+     * This function wraps around all rest requests
+     * If the request is authenticated, then it goes through
+     */
+    public RestHandler wrap(RestHandler original) {
+        return new RestHandler() {
+
+            @Override
+            public void handleRequest(RestRequest request, RestChannel channel, NodeClient client) throws Exception {
+                org.apache.logging.log4j.ThreadContext.clearAll();
+                if (checkAndAuthenticateRequest(request, channel, client)) {
+                    original.handleRequest(request, channel, client);
+                }
+            }
+        };
+    }
+
+    // True is authenticated, false if not - this is opposite of the Security plugin
+    private boolean checkAndAuthenticateRequest(RestRequest request, RestChannel channel, NodeClient client) throws Exception {
+        if (!authenticate(request, channel)) {
+            final OpenSearchException exc = new OpenSearchException("Authentication failed");
+            channel.sendResponse(new BytesRestResponse(channel, RestStatus.UNAUTHORIZED, exc));
+            return false;
+        }
+
+        if (threadContext.getHeader(ThreadContextConstants.OPENSEARCH_AUTHENTICATION_TOKEN_HEADER) == null) {
+            Map<String, String> jwtClaims = new HashMap<>();
+            jwtClaims.put("sub", "subject");
+            jwtClaims.put("iat", Instant.now().toString());
+            String encodedJwt = JwtVendor.createJwt(jwtClaims);
+            String prefix = "(nodeName="
+                + client.getLocalNodeId()
+                + ", requestId="
+                + request.getRequestId()
+                + ", path="
+                + request.path()
+                + ", jwtClaims="
+                + jwtClaims
+                + " checkAndAuthenticateRequest)";
+            log.info(prefix + " Created internal access token " + encodedJwt);
```

Comment:
Would logging all these info per request have an effect on performance? I'm not sure

Reply:
I can switch these log statements to debug, this was useful to see what was happening in the cluster but needs to be addressed before merge into main.

Reply:
Can you add a Todo here so we can track it?

Reply:
Changed to debug. Run the tests with `-Dtests.opensearch.logger.level=DEBUG` to see the output. This output helped with debugging the tests I wrote for this PR.